src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -164,6 +169,7 @@ void addConjunct(ref IsPatternTranslator self, BoundExpression expression)
                     case BoundDagPropertyEvaluation p:
                         {
                             var property = p.Property;
+                            property = property.TupleUnderlyingProperty ?? property;
```

Comment:
>property = property.TupleUnderlyingProperty ?? property; [](start = 28, length = 56)

This is probably unnecessary. At least I don't see us doing this for properties elsewhere.

Reply:
Addressed in https://github.com/dotnet/roslyn/pull/24059/commits/114a56e9077ed03f08443889b69cae6972a4acc6 of https://github.com/dotnet/roslyn/pull/24059